### PR TITLE
Finish JT.credentials functionality, add test

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Connect https://github.com/ansible/awx/issues/5171

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - AWX collection

##### AWX VERSION
```
8.0.0
```

### CODE REVIEW

File: awx_collection/plugins/modules/tower_job_template.py [hunk 1]
```diff
@@ -55,8 +55,17 @@
     credential:
       description:
         - Name of the credential to use for the job template.
+        - Deprecated, mutually exclusive with 'credentials'.
       version_added: 2.7
       type: str
+    credentials:
+      description:
+        - List of credentials to use for the job template.
+        - Will not remove any existing credentials.
```
Comment (chrismeyersfsu) [inline awx_collection/plugins/modules/tower_job_template.py]: Do want to consider the proposed batch attach/detatch .. or just consider that when it comes time?

File: awx_collection/plugins/modules/tower_job_template.py [hunk 2]
```diff
@@ -55,8 +55,17 @@
...
+        - Deprecated, mutually exclusive with 'credential'.
```
Comment (AlanCoding) [inline awx_collection/plugins/modules/tower_job_template.py]: oh, this was supposed to go on `vault_credential`, not here

File: awx_collection/plugins/modules/tower_job_template.py [hunk 3]
```diff
@@ -326,7 +336,14 @@ def main():
         state=dict(choices=['present', 'absent'], default='present'),
     )
 
-    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
+    module = TowerModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ('credential', 'credentials'),
```
Comment (chrismeyersfsu) [inline awx_collection/plugins/modules/tower_job_template.py:352]: Hot

**Review by chrismeyersfsu — APPROVED** [review]

File: awx_collection/plugins/modules/tower_job_template.py (hunk 1 above)
Comment (AlanCoding) [reply awx_collection/plugins/modules/tower_job_template.py]: By that you mean https://github.com/ansible/awx/issues/5245. We should pull @beeankha into this conversation.

The time frame for that will still be a while down the road. I would like to get this in user's hands now. I think it makes sense to punt on removal until that lands.

Comment (ryanpetrello) [reply awx_collection/plugins/modules/tower_job_template.py]: I agree that we shouldn't gate this change on the work @beeankha will be doing later for a bulk attach feature (though we could revisit this module at that point if we want to).

File: awx_collection/plugins/modules/tower_job_template.py [hunk 4]
```diff
@@ -36,7 +36,7 @@
       description:
         - The job type to use for the job template.
       required: True
-      choices: ["run", "check", "scan"]
```
Comment (ryanpetrello) [inline awx_collection/plugins/modules/tower_job_template.py:39]: 🗡

File: awx_collection/plugins/modules/tower_job_template.py (hunk 3 above)
Comment (ryanpetrello) [reply awx_collection/plugins/modules/tower_job_template.py:352]: 💯

**Review by ryanpetrello — APPROVED** [review]

**Pull request merged**